raiden/network/transport/matrix/client.py

```diff
@@ -251,31 +248,42 @@ def __init__(
         # Gets incremented every time a sync loop is completed. This is useful since the sync token
         # can remain constant over multiple loops (if no events occur).
         self.sync_iteration = 0
-
         self._sync_filter_id: Optional[int] = None
 
-    def create_sync_filter(self, broadcast_rooms: Dict[str, Room]) -> None:
-        ignore_rooms = [room.room_id for room in broadcast_rooms.values()]
-        filter_params = {
-            "room": {
-                # Filter out all room state events / messages from broadcast rooms
-                "not_rooms": ignore_rooms,
-                # Ignore "message recipts" from all rooms
-                "ephemeral": {"not_types": ["m.receipt"]},
-            },
-            # Get all presence updates
-            "presence": {"types": ["m.presence"]},
-        }
+    def create_sync_filter(
+        self, broadcast_rooms: Optional[Dict[str, Room]] = None, limit: Optional[int] = None
+    ) -> Optional[int]:
+        broadcast_room_filter: Dict[str, Any] = {"room": {}}
+        limit_filter: Dict[str, Any] = {"room": {}}
+        if broadcast_rooms:
+            ignore_rooms = [room.room_id for room in broadcast_rooms.values()]
+            broadcast_room_filter = {
+                "room": {
+                    # Filter out all room state events / messages from broadcast rooms
+                    "not_rooms": ignore_rooms,
+                    # Ignore "message receipts" from all rooms
+                    "ephemeral": {"not_types": ["m.receipt"]},
+                },
+                # Get all presence updates
+                "presence": {"types": ["m.presence"]},
+            }
+        if limit is not None:
+            limit_filter = {"room": {"timeline": {"limit": limit}}}
+
+        room_filter = {**(broadcast_room_filter["room"]), **(limit_filter["room"])}
+        sync_filter = {**broadcast_room_filter, **limit_filter}
+        sync_filter["room"] = room_filter
```

Comment:
This is quite hard to understand. 
We already have a `merge_dict` helper in `raiden.utils.datastructures` which could be used here.